src/shortcuts/__tests__/pinSubthoughts.ts

```diff
@@ -0,0 +1,41 @@
+import { inputHandlers } from '../../shortcuts'
+import { importText } from '../../action-creators'
+import { RANKED_ROOT } from '../../constants'
+import { noop } from 'lodash'
+import { createTestStore } from '../../test-helpers/createTestStore'
+import { attributeEquals } from '../../selectors'
+
+it('toggle =pinChildren attribute of parent of cursor', async () => {
+
+  const store = createTestStore()
+
+  const { keyDown } = inputHandlers(store)
+
+  // import thoughts
+  await store.dispatch(importText(RANKED_ROOT, `
+  - a
+    - b
+      - c
+      - d
+    - e
+      - f
+      - g
+`))
+
+  await store.dispatch({ type: 'setCursor', thoughtsRanked: [
+    { value: 'a', rank: '0' },
+    { value: 'b', rank: '1' },
+  ] })
+
+  // toggle =pinChildren attribute
+  keyDown({ preventDefault: noop, key: 'p', altKey: true, shiftKey: true } as KeyboardEvent)
+
+  // parent of cursor should have =pinChildren set to true
+  expect(attributeEquals(store.getState(), ['a'], '=pinChildren', 'true')).toBeTruthy()
+
+  // toggle =pinSubthoughts attribute
+  keyDown({ preventDefault: noop, key: 'p', altKey: true, shiftKey: true } as KeyboardEvent)
+
+  // parent of cursor should not have =pinChildren attribute
+  expect(attributeEquals(store.getState(), ['a'], '=pinChildren', 'false')).toBeFalsy()
```

Comment:
In general I recommend breaking up assertions for different behaviors into separate tests. In this test, if toggle on fails, then it will never execute the assertion for toggle off. It would be more informative to have a separate test for each. Technically we should also have a test for toggle on from `=pinChildren/false`, not just from the initial state (no `=pinChildren`).

Another area of improvement in this test is the implicit test of the keyboard shortcut. It would be better to import the shortcut directly and call `exec`. We do not need to test the keyboard shortcut itself as we don't want this test to fail if there is something wrong with `keyDown`. That should be tested separately in a shortcut test if at all.

Reply:
Sure. It makes sense.